src/test/tests_common.h

```diff
@@ -62,12 +62,25 @@ struct test_options {
 	int	self_rank;
 	int	mypid;
 	int	is_server;
+	int	num_attach_retries;
+	int	do_assert;
 };
 
 struct test_options opts;
 
 int g_shutdown;
 
+void
+tc_test_init(int rank, int pid, int is_server, int num_attach_retries,
+	     int do_assert)
+{
+	opts.self_rank		= rank;
```

Comment:
one thing to do here would be to add new field to 'opts' and call it 'is_initialized' or some such and set it to true.
In all other tc_* calls we can then D_ASSERTF(opts.is_initialized == true, "Should call tc_test_init first");

This will allow us to catch situations of missing tc_test_init()